internal/sql/sql.go

```diff
@@ -167,7 +167,7 @@ LOOP:
 			}
 		}
 		if !locked {
-			break LOOP
+			return err
```

Comment:
Returning here avoids printing an error message that mentions "transaction attempt x" for all cases where the transaction wouldn't be retried anyway. This has been bothering me for some time, we sometimes even expose it as a 500 response.